src/call.js

```diff
@@ -1,14 +1,146 @@
+import {
+  addToPresenceList,
+  hidePresence,
+  setupCreateButton,
+  showCall,
+  showEntry,
+  showPresence,
+} from './dom.js';
+
 window.addEventListener('DOMContentLoaded', () => {
-  const container = document.getElementById('container');
+  setupCreateButton(createRoom);
+
+  // If room URL and name params were provided, join that room
+  const params = new URLSearchParams(window.location.search);
+  const roomURL = params.get('roomURL');
+  const roomName = params.get('roomName');
+  if (roomURL && roomName) {
+    joinRoom(roomURL, roomName);
+    return;
+  }
+
+  // If room URL and name params were not provided, show
+  // entry element (with call creation button)
+  showEntry();
+});
+
+/**
+ * Creates and joins a Daily room
+ * @returns {Promise<void>}
+ */
+async function createRoom() {
+  const url = `/.netlify/functions/createRoom`;
+  const errMsg = 'failed to create room';
 
-  callFrame = window.DailyIframe.createFrame(container, {
+  // Create Daily room
+  let res;
+  try {
+    res = await fetch(url);
+  } catch (e) {
+    console.error('failed to make Daily room creation request', e);
+  }
+
+  // Retrieve body, if it exists in the response
+  let body;
+  try {
+    body = await res.json();
+  } catch (e) {
+    console.warn('No body present in room creation response');
+  }
+
+  // Check status code
+  if (res.status !== 200) {
+    const msg = `${errMsg}. Status code: ${res.status}`;
+    console.error(`failed to create Daily room: ${msg}`, body);
+    return;
+  }
+  joinRoom(body.url, body.name);
+}
+
+/**
+ * Joins a Daily room and retrieves its existing participants
+ * @param roomURL
+ * @param roomName
+ */
+function joinRoom(roomURL, roomName) {
+  showCall();
+
+  // Set up the call frame and join the call
+  const callContainer = document.getElementById('callFrame');
+  const callFrame = window.DailyIframe.createFrame(callContainer, {
     showLeaveButton: true,
-    iframeStyle: {
-      position: 'fixed',
-      width: 'calc(100% - 1rem)',
-      height: 'calc(100% - 5rem)',
-    },
   });
-  // TODO: Replace the following URL with your own room URL.
-  callFrame.join({ url: 'https://<your-domain>.daily.co/<room-name>' });
-});
+  callFrame.on('joined-meeting', () => {
+    // Presence view is no longer needed once the participant is in the room.
+    hidePresence();
+  });
+  callFrame.join({ url: roomURL });
+
+  updateInviteURL(roomURL, roomName);
+
+  // Fetch existing participants and show them next to the call frame
+  fetchParticipants(roomName);
+}
+
+function updateInviteURL(roomURL, roomName) {
+  const inviteURLEle = document.getElementById('inviteURL');
```

Comment:
Small nit for the invite URL: it's hard to tell what the actual Daily room URL is. Like, if I want others to join the room to test this, it doesn't need to be through the demo view, it can just be the Daily room. My suggestion would be to include `Daily room: <url>` under `Invite URL: <url>` in the UI, or just the Daily room URL... both is probably the nicer version in case people do want to stay in the demo view for every participant.

Reply:
Good idea, updating